src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadYamlStep.java

```diff
@@ -60,20 +60,36 @@
  */
 public class ReadYamlStep extends AbstractFileOrTextStep {
 
+	public static final int LIBRARY_DEFAULT_MAX_ALIASES_FOR_COLLECTIONS = new LoaderOptions().getMaxAliasesForCollections();
 	public static final String DEFAULT_MAX_ALIASES_PROPERTY = ReadYamlStep.class.getName() + ".DEFAULT_MAX_ALIASES_FOR_COLLECTIONS";
 	@SuppressFBWarnings(value={"MS_SHOULD_BE_FINAL"}, justification="Non final so that an admin can adjust the value through the groovy script console without restarting the instance.")
-	public static /*almost final*/ int DEFAULT_MAX_ALIASES_FOR_COLLECTIONS = Integer.getInteger(DEFAULT_MAX_ALIASES_PROPERTY, -1);
+	private static /*almost final*/ int DEFAULT_MAX_ALIASES_FOR_COLLECTIONS = setDefaultMaxAliasesForCollections(Integer.getInteger(DEFAULT_MAX_ALIASES_PROPERTY, -1));
 	public static final String MAX_MAX_ALIASES_PROPERTY = ReadYamlStep.class.getName() + ".MAX_MAX_ALIASES_FOR_COLLECTIONS";
-	@SuppressFBWarnings(value={"MS_SHOULD_BE_FINAL"}, justification="Non final so that an admin can adjust the value through the groovy script console without restarting the instance.")
-	public static /*almost final*/ int MAX_MAX_ALIASES_FOR_COLLECTIONS = Integer.getInteger(MAX_MAX_ALIASES_PROPERTY, 1000);
+	public static final int MAX_MAX_ALIASES_FOR_COLLECTIONS = Integer.getInteger(MAX_MAX_ALIASES_PROPERTY, LIBRARY_DEFAULT_MAX_ALIASES_FOR_COLLECTIONS);
```

Comment:
By making this final you remove the option for an administrator to set this value via the script console.

Reply:
Correct. I think this is what @daniel-beck requested in the target PR.

Reply:
Sorry, btw. I missed your comment somehow. Would have replied sooner had I seen it.

Reply:
Here -> https://github.com/jenkinsci/pipeline-utility-steps-plugin/pull/152#issuecomment-1261988409

Reply:
I don't see Daniel suggesting that anywhere, just your interpretation of what he requested :shrug: 
IMO at least having a system property and enabling setting it via the script console is the same level of "inconvenience" that Daniel was after. The latter is just allowing you to fix the plane while it is flying :)

Reply:
> I don't think it's a good idea to let admins make terrible decisions too easily, as that ultimately reflects badly on Jenkins when they do. System property is probably fine, UI option would be "too easy".

I am interpreting "UI option" as the script console.

@daniel-beck?

Reply:
And this is just for MAX_MAX..., not for DEFAULT_MAX...

Reply:
UI option would be a button. Script console usage is nuts anyway, if you go this route, you're on your own.

Reply:
Hi everyone, do we have a consensus on this?

Reply:
@daniel-beck from your comment I read 

* non-finality for script console use is generally nice
* should not be allowed in the case of the `MAX_MAX_ALIASES_FOR_COLLECTIONS` because making it harder to change means the admin has to make a "more deliberate decision" to do so rather than simply run a script.

@rsandell - are you okay with this?

Reply:
A good solution for me would be:

1. Non-`final` `MAX_MAX`.
2. Ignore what I wrote in the other PR and brought up again here.

I think that makes this PR unnecessary because that's how it currently works AFAIUI.

A better solution for me would be:

1. Non-`final` `MAX_MAX`.
2. Access it through a `Math#min` filter with hardcoded large but reasonable limit (500? 1000? Might take some testing) so that admins cannot exceed that even by setting `MAX_MAX` unreasonably large.

I hope this clarifies things.

Reply:
Ok, I can revert the change making `MAX_MAX` `final`.

Wrt the other change, @rsandell you will ultimately be merging this so which do you prefer.

Reply:
I can also just close this and we keep the original.